src/test/java/mate/academy/springbootintro/controller/BookControllerTest.java

```diff
@@ -38,15 +38,38 @@
 import org.springframework.test.web.servlet.MvcResult;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
 import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
 
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class BookControllerTest {
 
+    protected static MockMvc mockMvc;
     private static final Long BOOK_ID_1 = 1L;
     private static final Long BOOK_ID_2 = 2L;
     private static final Long BOOK_ID_3 = 3L;
-
-    protected static MockMvc mockMvc;
+    private static final String SPRING_DATASOURCE_URL =
+            "spring.datasource.url";
+    private static final String SPRING_DATASOURCE_USERNAME =
+            "spring.datasource.username";
+    private static final String SPRING_DATASOURCE_PASSWORD =
+            "spring.datasource.password";
+    private static final String SPRING_DATASOURCE_DRIVER_CLASS_NAME =
+            "spring.datasource.driver-class-name";
+    private static final String CATEGORY_NAME = "magic";
+    private static final String CATEGORY_DESCRIPTION = "dragons";
+    private static final String BOOK_TITLE_1 = "Rekrut";
+    private static final String BOOK_AUTHOR_1 = "Vanbord";
+    private static final String BOOK_COVER_IMAGE_1 = "https://randomImage1.jpg";
+    private static final String BOOK_DESCRIPTION_1 = "great book";
+    private static final BigDecimal BOOK_PRICE_1 = BigDecimal.TEN;
+    private static final String BOOK_TITLE_2 = "Harry Potter";
+    private static final String BOOK_AUTHOR_2 = "Rowling";
+    private static final String BOOK_ISBN_2 = "123456789";
+    private static final BigDecimal BOOK_PRICE_2 = BigDecimal.valueOf(19.99);
+    private static final String BOOK_COVER_IMAGE_2 = "https://randomImage1.jpg";
+    private static final String BOOK_DESCRIPTION_2 = "great book";
+    private static final String BOOK_SEARCH_TITLE = "Harry Potter";
+    private static final String BOOK_SEARCH_AUTHOR = "Rowling";

```

Comment:
I don't know exactly what your book model looks like, but you certainly don't need unique values in so many places, you can use the same constant in several places.